packages/@glimmer/validator/lib/debug.ts

```diff
@@ -218,31 +194,23 @@ if (DEBUG) {
 
     if (!transaction) return;
 
-    let currentTransaction = TRANSACTION_STACK[TRANSACTION_STACK.length - 1];
-
-    if (currentTransaction.deprecate) {
-      deprecate(makeTrackingErrorMessage(transaction, obj, keyName), false, {
-        id: 'autotracking.mutation-after-consumption',
-      });
-    } else {
-      // This hack makes the assertion message nicer, we can cut off the first
-      // few lines of the stack trace and let users know where the actual error
-      // occurred.
-      try {
-        assert(false, makeTrackingErrorMessage(transaction, obj, keyName));
-      } catch (e) {
-        if (e.stack) {
-          let updateStackBegin = e.stack.indexOf('Stack trace for the update:');
-
-          if (updateStackBegin !== -1) {
-            let start = nthIndex(e.stack, '\n', 1, updateStackBegin);
-            let end = nthIndex(e.stack, '\n', 4, updateStackBegin);
-            e.stack = e.stack.substr(0, start) + e.stack.substr(end);
-          }
+    // This hack makes the assertion message nicer, we can cut off the first
```

Comment:
not related to this PR, but I wonder what is going on here -- why are we cutting of parts of the error message? are those parts ever needed?